File: Guess_number.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int main() {
    srand(time(0)); 
    int secretNumber = rand() % 100 + 1; 
    int attempts = 0;
    int guess;

    cout << "Welcome to the Guessing Game!" << endl;
    cout << "I have selected a random number between 1 and 100. Try to guess it!" << endl;

    while (true) {
        cout << "Enter your guess: ";
        cin >> guess;
        attempts++;

        if (guess == secretNumber) {
            cout << "Congratulations! You guessed the secret number " << secretNumber << " in " << attempts << " attempts." << endl;
            break;
        } else if (guess < secretNumber) {
            cout << "Too low! Try again." << endl;
        } else {
            cout << "Too high! Try again." << endl;
        }

        if (attempts >= 10) {
            cout << "Sorry, you've reached the maximum number of attempts. The secret number was " << secretNumber << "." << endl;
            break;
        }
    }

    return 0;
}

File: GuessNumber.js
const randomNumber = Math.floor(Math.random() * 100) + 1;
let attempts = 0;

while (true) {
    const guess = prompt('Guess a number between 1 and 100:');
    attempts++;

    if (guess === null) {
        alert('Game over. You quit.');
        break;
    }

    const guessNum = parseInt(guess);

    if (isNaN(guessNum) || guessNum < 1 || guessNum > 100) {
        alert('Please enter a valid number between 1 and 100.');
    } else if (guessNum === randomNumber) {
        alert(`Congratulations! You guessed the correct number (${randomNumber}) in ${attempts} attempts.`);
        break;
    } else if (guessNum < randomNumber) {
        alert('Try a higher number.');
    } else {
        alert('Try a lower number.');
    }
}